### PR TITLE
Add editor for draft decks

### PR DESCRIPTION
**Pull request by ystros**
Progress toward https://github.com/throneteki/throneteki/issues/3206

### CODE REVIEW

File: Components/Decks/DeckSummary.jsx [hunk 1]
```diff
@@ -17,116 +17,34 @@ class DeckSummary extends React.Component {
...
+                    onCardMouseOver={ this.onCardMouseOver } />
                 <div className='col-xs-12 no-x-padding'>
-                    <div className='cards'>
-                        { cardsToRender }
-                    </div>
+                    <CardTypeGroups
+                        cards={ this.props.deck.plotCards.concat(this.props.deck.drawCards) }
+                        onCardMouseOut={ this.onCardMouseOut }
+                        onCardMouseOver={ this.onCardMouseOver }
+                        useScheme={ this.props.deck.agenda && this.props.deck.agenda.code === '05045' } />
```
Comment (EvadLhorg) [inline Components/Decks/DeckSummary.jsx:47]: useScheme is missing an "s" as the prop name in CardTypeGroups is useSchemes

File: Components/Decks/DraftDeckEditor.jsx [hunk 2]
```diff
@@ -0,0 +1,428 @@
...
+                        <div className='form-group'>
+                            <label className='col-sm-12'>Deck:</label>
+                        </div>
+                        <div className='col-sm-12'>
+                            <CardTypeGroups
+                                cards={ this.state.plotCards.concat(this.state.drawCards) }
+                                onCardClick={ card => this.handleRemoveCard(card.code) }
+                                onCardMouseOut={ () => this.clearHoverCard() }
+                                onCardMouseOver={ card => this.updateHoverCard(card) }
+                                useScheme={ this.state.agenda && this.state.agenda.code === '05045' } />
```
Comment (EvadLhorg) [inline Components/Decks/DraftDeckEditor.jsx:386]: useScheme is missing an "s" as the prop name in CardTypeGroups is useSchemes

File: Components/Decks/DraftDeckEditor.jsx [hunk 3]
```diff
@@ -0,0 +1,428 @@
...
+                    </div>
+                </div>
+                <div className='row'>
+                    <div className='col-sm-5'>
+                        <div className='form-group'>
+                            <label className='col-sm-12'>Remaining Drafted Cards:</label>
+                        </div>
+                        <div className='col-sm-12'>
+                            <CardTypeGroups
+                                cards={ this.state.remainingCards }
```
Comment (EvadLhorg) [inline Components/Decks/DraftDeckEditor.jsx:366]: this CardTypeGroups shows all remaining cards including agendas. Maybe we have to find some way to filter out agendas from the clickable cards as those are already in the agenda drop down
grafik

File: Components/Decks/DraftDeckEditor.jsx (hunk 2 above)
Comment (ystros) [reply Components/Decks/DraftDeckEditor.jsx]: Fixed!

**Pull request merged**